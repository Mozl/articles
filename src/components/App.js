import React from 'react';
import Nav from './Nav';
import ArticleContainer from './ArticleContainer';
import { ArticleProvider } from '../contexts/ArticleContext';
import { BrowserRouter as Router } from 'react-router-dom';

const App = () => {
  const slowConnection = window.setTimeout(() => {
    alert("the page is taking its sweet time loading");
  }, 5000);

  window.addEventListener('load', () => {
      window.clearTimeout(slowConnection);
  }, false);
  const offlineWarning = () => {
    alert("No network connection")
  }
  window.addEventListener('offline', offlineWarning);
  return (
    <ArticleProvider>
      <Router>
        <div className="App">
          <Nav />
          <ArticleContainer as={ArticleContainer} />
        </div>
      </Router>
    </ArticleProvider>
  );
}

export default App;
