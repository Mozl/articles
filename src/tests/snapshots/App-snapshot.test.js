import React from 'react';
import App from '../../components/App';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});