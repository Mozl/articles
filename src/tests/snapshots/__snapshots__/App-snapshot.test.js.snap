// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <nav
    className="sc-bdVaJa gSibpX"
    id="nav"
  >
    <a
      href="/article1"
      id="link"
      onClick={[Function]}
    >
      <img
        alt="Article 1"
        height="30"
        id="home-link"
        src="BBC_logo_white.png"
        tabIndex="1"
        width="100"
      />
    </a>
    <a
      href="/article2"
      id="link"
      onClick={[Function]}
    >
      <li
        className="sc-bwzfXH czeCUF"
        id="article2-link"
        tabIndex="2"
      >
        Article 2
      </li>
    </a>
    <a
      href="/article3"
      id="link"
      onClick={[Function]}
    >
      <li
        className="sc-bwzfXH czeCUF"
        id="article3-link"
        tabIndex="3"
      >
        Article 3
      </li>
    </a>
    <a
      href="/article4"
      id="link"
      onClick={[Function]}
    >
      <li
        className="sc-bwzfXH czeCUF"
        id="article4-link"
        tabIndex="4"
      >
        Article 4
      </li>
    </a>
    <a
      href="/article5"
      id="link"
      onClick={[Function]}
    >
      <li
        className="sc-bwzfXH czeCUF"
        id="article5-link"
        tabIndex="5"
      >
        Article 5
      </li>
    </a>
    <a
      href="/ranking"
      id="link"
      onClick={[Function]}
    >
      <li
        className="sc-bwzfXH czeCUF"
        id="ranking-link"
        tabIndex="6"
      >
        Ranking
      </li>
    </a>
  </nav>
  <div
    className="sc-htoDjs jYbOhy"
  />
</div>
`;
